services/scraper/service.go

```diff
@@ -42,13 +49,23 @@ type Service struct {
 
 	discoverers []Discoverer
 
-	// TargetManager represents a scraping/discovery manager
-	mgr interface {
-		ApplyConfig(cfg *config.Config) error
-		Stop()
-		Start()
```

Comment:
The prometheus fork we had was just to split `Run` into `Start` and `Wait`. But the latest version of prometheus looks like it should support calling `Stop` before `Run` in racy cases, so I don't think we need the fork any longer.

Reply:
awesome!